src/txadmission.cpp

```diff
@@ -327,22 +351,27 @@ void CommitTxToMempool()
         // from re-requests.
         LOG(MEMPOOL, "popping txdeferQ, size %d\n", txDeferQ.size());
         // this could be a lot more efficient
-        while (!txDeferQ.empty())
+        uint64_t count = 0;
+        uint64_t maxmove = max(avgCommitBatchSize * 2, minCommitBatchSize);
+        while ((!txDeferQ.empty()) && (count < maxmove))
         {
+            count++;
             const uint256 &hash = txDeferQ.front().tx->GetHash();
             mapWasDeferred.emplace(hash, txDeferQ.front());
-
             txDeferQ.pop();
         }
     }
 
     if (!mapWasDeferred.empty())
-        LOG(MEMPOOL, "%d tx were deferred\n", mapWasDeferred.size());
+        LOG(MEMPOOL, "Enqueueing %d deferred tx\n", mapWasDeferred.size());
 
-    for (auto &it : mapWasDeferred)
     {
-        LOG(MEMPOOL, "attempt enqueue deferred %s\n", it.first.ToString());
-        EnqueueTxForAdmission(it.second);
+        LOCK(csTxInQ);
+        for (auto &it : mapWasDeferred)
+        {
+            // LOG(MEMPOOL, "attempt enqueue deferred %s\n", it.first.ToString());
```

Comment:
why is the `LOG` statement commented?

Reply:
its spammy unless you are specifically debugging the queueing